dlite_entities_service/models/soft.py

```diff
@@ -0,0 +1,148 @@
+            alias="type",
+            description="The type of the described property, e.g., an integer.",
+        ),
+    ]
+    shape: Annotated[
+        list[str] | None,
+        Field(
+            description=(
+                "The dimension of multi-dimensional properties. This is a list of "
+                "dimension expressions referring to the dimensions defined above. For "
+                "instance, if an entity have dimensions with names `H`, `K`, and `L` "
+                "and a property with shape `['K', 'H+1']`, the property of an instance "
+                "of this entity with dimension values `H=2`, `K=2`, `L=6` will have "
+                "shape `[2, 3]`."
+            ),
+            validation_alias=AliasChoices("dims", "shape"),
+        ),
+    ] = None
+    unit: Annotated[str | None, Field(description="The unit of the property.")] = None
+    description: Annotated[
+        str, Field(description="A human-readable description of the property.")
+    ]
+
+
+class SOFTEntity(BaseModel):
+    """A minimum Field set SOFT Entity to be used in a versioned SOFT Entity model."""
+
+    model_config = ConfigDict(extra="forbid")
+
+    name: Annotated[str | None, Field(description="The name of the entity.")] = None
+    version: Annotated[
+        str | None, Field(description="The version of the entity.")
+    ] = None
+    namespace: Annotated[
+        AnyHttpUrl | None, Field(description="The namespace of the entity.")
+    ] = None
+    uri: Annotated[
+        AnyHttpUrl | None,
+        Field(
+            description=(
+                "The universal identifier for the entity. This MUST start with the base"
+                " URL."
+            ),
+        ),
+    ] = None
+    description: Annotated[str, Field(description="Description of the entity.")] = ""
+
+    @field_validator("uri", "namespace", mode="after")
+    @classmethod
+    def _validate_base_url(cls, value: AnyHttpUrl) -> AnyHttpUrl:
+        """Validate `uri` and `namespace` starts with the current base URL for the
+        service."""
+        if not str(value).startswith(str(CONFIG.base_url)):
+            error_message = (
+                "This service only works with SOFT entities at " f"{CONFIG.base_url}.\n"
+            )
+            raise ValueError(error_message)
+        return value
+
+    @field_validator("uri", mode="after")
+    @classmethod
+    def _validate_uri(cls, value: AnyHttpUrl) -> AnyHttpUrl:
+        """Validate `uri` is consistent with `name`, `version`, and `namespace`."""
+        if URI_REGEX.match(str(value)) is None:
+            error_message = (
+                "The 'uri' is not a valid SOFT7 entity URI. It must be of the form "
+                f"{str(CONFIG.base_url).rstrip('/')}/{{version}}/{{name}}.\n"
+            )
+            raise ValueError(error_message)
+        return value
+
+    @model_validator(mode="before")
+    @classmethod
+    def _check_cross_dependent_fields(cls, data: Any) -> Any:
```

Comment:
_cross_check_dependent_fields seems especially long for something that is relatively simple. Cross check also implies something more complex (e.g that the name and namespace are consistent with the version according to a complex rule or something)

Reply:
It should be noted that the content and logic of the validators have not been changed. They have merely been moved around a bit to where they not belong. However, it may still be fine to address in this PR, for sure :)

Reply:
Essentially, `namespace+version+name` in that specific order MUST be equal to `uri` if all of these fields are given. That is what's being checked :)